core/domain/exp_jobs_one_off.py

```diff
@@ -685,3 +689,90 @@ def reduce(status, values):
             yield (status, len(values))
         else:
             yield (status, values)
+
+
+class ValidationOfImagesOnGCS(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for checking that all the images in the GAE are there in
+        the GCS or not.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileModel]
+
+    @staticmethod
+    def map(file_model):
+        # This job is allowed to run only in Production environment since it
+        # uses GcsFileSystem which can't be used in Development environment.
+        if not feconf.DEV_MODE:
+            instance_id = file_model.id
+            filetype = instance_id[instance_id.rfind('.') + 1:]
+            # To separate the image entries from the audio entries we get from
+            # the FileSnapshotContentModel.
+            if filetype in ALLOWED_IMAGE_EXTENSIONS:
+                pattern = re.compile(
```

Comment:
So much duplication. Can we put this regex somewhere in a common constant?

Reply:
Done!